File: app/src/main/java/se/gustavkarlsson/skylight/android/dagger/modules/replaceable/SettingsModule.java
package se.gustavkarlsson.skylight.android.dagger.modules.replaceable;

import dagger.Binds;
import dagger.Module;
import dagger.Reusable;
import se.gustavkarlsson.skylight.android.settings.Settings;
import se.gustavkarlsson.skylight.android.settings.SharedPreferencesSettings;

@Module
public abstract class SettingsModule {

	// Published
	@Binds
	@Reusable
	abstract Settings bindSettings(SharedPreferencesSettings sharedPreferencesSettings);
}

File: app/src/main/kotlin/se/gustavkarlsson/skylight/android/di/modules/SettingsModule.kt
package se.gustavkarlsson.skylight.android.di.modules

import com.f2prateek.rx.preferences2.RxSharedPreferences
import se.gustavkarlsson.skylight.android.services.Settings

interface SettingsModule {
	val rxSharedPreferences: RxSharedPreferences
	val settings: Settings
}
